app/src/main/java/net/gsantner/markor/ui/NewFileDialog.java

```diff
@@ -94,7 +94,7 @@ public Dialog onCreateDialog(Bundle savedInstanceState) {
         final Spinner templateSpinner = root.findViewById(R.id.new_file_dialog__template);
         final String[] typeSpinnerToExtension = getResources().getStringArray(R.array.new_file_types__file_extension);
 
-        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT && appSettings.hasPasswordBeenSetOnce()) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M && !TextUtils.isEmpty(appSettings.getPassword())) {
```

Comment:
Just checking if password exists. This avoids issues with `hasPasswordBeenSetOnce` being out of sync with actual status after restore of backup

Reply:
@gsantner Sure that this will work?

Reply:
It seems to work fine for me. Is there a specific bug you anticipate / see here?

Reply:
Why not simply add a method appSettings.isPasswortSet()? Then the information how it is stored is encapsulate at one place. We could have an extra-field or check if the char[] is null.